File: app/src/main/java/com/veganbeauty/admin/data/repository/CustomerRepository.java
package com.veganbeauty.admin.data.repository;

import androidx.lifecycle.LiveData;
import com.veganbeauty.admin.data.local.dao.CustomerDao;
import com.veganbeauty.admin.data.local.entities.CustomerEntity;
import com.veganbeauty.admin.data.remote.FirebaseService;

import java.util.List;

public class CustomerRepository {
    private final CustomerDao customerDao;
    private final FirebaseService firebaseService;

    public CustomerRepository(CustomerDao customerDao, FirebaseService firebaseService) {
        this.customerDao = customerDao;
        this.firebaseService = firebaseService;
    }

    public LiveData<List<CustomerEntity>> getAllCustomers() {
        return customerDao.getAllLiveData();
    }

    public void syncFromFirebase() {
        List<CustomerEntity> remoteList = firebaseService.fetchAllCustomers();
        if (!remoteList.isEmpty()) {
            customerDao.insertAllSync(remoteList);
        }
    }
}

File: app/src/main/java/com/veganbeauty/admin/data/repository/CustomerRepository.kt
package com.veganbeauty.admin.data.repository

import androidx.lifecycle.LiveData
import com.veganbeauty.admin.data.local.dao.CustomerDao
import com.veganbeauty.admin.data.local.entities.CustomerEntity
import com.veganbeauty.admin.data.remote.FirebaseService
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext

class CustomerRepository(
    private val customerDao: CustomerDao,
    private val firebaseService: FirebaseService
) {

    val allCustomers: LiveData<List<CustomerEntity>> = customerDao.getAllLiveData()

    suspend fun syncFromFirebase() = withContext(Dispatchers.IO) {
        val remoteList = firebaseService.fetchAllCustomers()
        if (remoteList.isNotEmpty()) {
            customerDao.insertAllSync(remoteList)
        }
    }
}
